pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -102,3 +151,22 @@ func (n Node) hasCompatibleResources(resourceList v1.ResourceList, it cloudprovi
 	}
 	return true
 }
+
+func (n *Node) recalculateDaemonResources() {
+	n.daemonResources = nil
+	for _, daemon := range n.daemons {
+		podRequirements := v1alpha5.NewPodRequirements(daemon)
```

Comment:
This is recalculating daemon overhead on every iteration. Can we maybe take in a map[pod]Requirements to avoid recomputing? We could also build this in the NewNode constructor.

Reply:
Yeah, I'll just compute in the constructor so I can reuse it.